src/npc.cpp

```diff
@@ -2427,8 +2430,8 @@ void npc::reboot()
     path.clear();
     last_player_seen_pos = cata::nullopt;
     last_seen_player_turn = 999;
-    wanted_item_pos = no_goal_point;
-    guard_pos = no_goal_point;
+    wanted_item_pos = tripoint_min;
+    guard_pos = tripoint_min;
```

Comment:
If you're changing these two, why not `goal = no_goal_point` ?

Reply:
The three have different types now.  I kept `goal = no_goal_point` because it was the correct type, but I couldn't keep the others the same.  I could have invented new constants for them, but they should really just be migrated to `optional`, so this was a minimum-effort change.